### PR TITLE
build: add precommit hook

### PR DESCRIPTION
**Pull request by JayKan**
## Summary and Context
- Update to node `v8` for Yarn CI (Travis)
- Updated `prettier` & `eslint` configs 
- Add precommit hook for `prettier`

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,6 +80,17 @@
...
     "react-test-renderer": "^16.4.0"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,json}": [
+      "prettier --write",
```
Comment (coleturner) [inline package.json]: Can we add `eslint --fix` after Prettier step? :)

File: package.json [hunk 2]
```diff
@@ -115,13 +125,5 @@
       "node",
       "mjs"
     ]
-  },
-  "babel": {
```
Comment (coleturner) [inline package.json]: Why remove Babel config?

File: package.json (hunk 1 above)
Comment (JayKan) [reply package.json]: Sounds good 👍

File: package.json (hunk 2 above)
Comment (JayKan) [reply package.json]: We already have the `.babelrc` file, I don't think we will need it within our `package.json`

Comment (coleturner) [reply package.json]: Ahhh right, did not observe that - good catch

**Comment by JayKan** [comment]
@coleturner, see https://github.com/aria-51/aria-lights-out/pull/20/commits/d6233f36b44c4d0f7170c146a6076506fbd176d9 for changes

File: package.json [hunk 3]
```diff
@@ -81,6 +81,18 @@
...
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,mjs,json}": [
+      "prettier --write",
+      "npm run lint",
```
Comment (coleturner) [inline package.json]: `eslint --fix .` would be better, since it would tap into the binary directly

**Pull request merged**